core/store/models/log_events.go

```diff
@@ -88,30 +88,42 @@ func TopicFiltersForRunLog(logTopics []common.Hash, jobID *ID) [][]common.Hash {
 func FilterQueryFactory(i Initiator, from *big.Int) (ethereum.FilterQuery, error) {
 	switch i.Type {
 	case InitiatorEthLog:
-		return newInitiatorFilterQuery(i, from, nil), nil
+		if from == nil {
+			from = i.InitiatorParams.FromBlock.ToInt()
+		} else if from != nil && i.InitiatorParams.FromBlock != nil {
+			from = utils.MaxBigs(from, i.InitiatorParams.FromBlock.ToInt())
+		}
+		q := ethereum.FilterQuery{
+			FromBlock: from,
+			ToBlock:   i.InitiatorParams.ToBlock.ToInt(),
+			Addresses: utils.WithoutZeroAddresses([]common.Address{i.Address}),
+			Topics:    i.Topics,
+		}
+		return q, nil
+
 	case InitiatorRunLog:
 		topics := []common.Hash{RunLogTopic20190207withoutIndexes, RunLogTopic20190123withFullfillmentParams, RunLogTopic0original}
 		filters := TopicFiltersForRunLog(topics, i.JobSpecID)
-		return newInitiatorFilterQuery(i, from, filters), nil
+		return ethereum.FilterQuery{
```

Comment:
Looks like you could assemble this filter query at the top of the function and just fill in `ToBlock` under `InitiatorEthLog`.